.circleci/initialize-manager.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+import traceback
+
+from fabric.api import *
+
+from common import *
+# This is expected to be launch from the ci container
+from ci_variables import *
+
+def initialize_manager(max_runtime):
+    """ Preforms the prerequisite tasks for all CI jobs that will run on the manager instance
+
+    max_runtime (seconds): The maximum uptime this manager should have be for it is terminated
+        This covers potential livelock scenario in one of the jobs (that produces fs activity)
+
+    filesystem_timeout (seconds):  If no fs activity is detected in ~/firesim after 2x this
+        interval, the manager is terminated.  CircleCI provides no mechanism for
+        doing cleanup after all jobs are done (even if some of those jobs fail).
+        So here we use FS activity as a proxy for doneness
+    """
+
+    # Catch any exception that occurs so that we can gracefully teardown
+    try:
+        put(ci_workdir + "/scripts/machine-launch-script.sh", manager_home_dir)
+        with cd(manager_home_dir):
+            run("chmod +x machine-launch-script.sh")
+            run("sudo ./machine-launch-script.sh")
+            run("git clone https://github.com/firesim/firesim.git")
+
+        with cd(manager_fsim_dir):
+            run("git checkout " + ci_commit_sha1)
+            run("./build-setup.sh --fast")
+
+        with cd(manager_fsim_dir), prefix("source ./sourceme-f1-manager.sh"):
+            run(".circleci/firesim-managerinit.expect {} {} {}".format(
```

Comment:
FYI we can simplify the `expect` here by just using a configuration file and having this expect just click enter: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html.

Reply:
Yeah that's much clearner. Though I think we should just split out the one-time aws configure from managerinit, as @timsnyder-siv has previously suggested (IIRC),  and then eventually drop the expect altogether.